pkg/controller/clusterstorage/clusterstorage_controller.go

```diff
@@ -336,6 +363,21 @@ func newStorageClassForCluster(infrastructure *configv1.Infrastructure) (*storag
 	}
 }
 
+// Examines the default storage class and compares it to the one defined in the assets directory.
+// If changes are found in a configurable attribute, update the existing StorageClass and mark it
+// for update.
+func updateStorageClass(existingClass *storagev1.StorageClass, sc *storagev1.StorageClass) bool {
+	needUpdate := false
+	if existingClass.AllowVolumeExpansion == nil ||
+		*existingClass.AllowVolumeExpansion != *sc.AllowVolumeExpansion ||
+		!reflect.DeepEqual(existingClass.MountOptions, sc.MountOptions) ||
+		!reflect.DeepEqual(existingClass.AllowedTopologies, sc.AllowedTopologies) {
+		needUpdate = true
+	}
+
+	return needUpdate
+}
+
 func getRelatedObjects(sc *storagev1.StorageClass) []configv1.ObjectReference {
```

Comment:
we typically do this by replacing a field that can be changed (annotations in this case) and then comparing the two objects. so something like:

```
newSCCopy := newSC.DeepCopy()
newSCCopy.Annotation = oldSC.Annotation

if !reflect.DeepEqual(newSCCopy, oldSC) {
  returns true
}
return false
````

Reply:
ah this comparsion might not work for SC because it does not have spec. I do not like having to hardcode fields and we will need some tests.

Reply:
I think we should preserve all fields from `ObjectMeta` type rather than just annotations. So something like:

```golang
newsc.ObjectMeta =oldSC.ObjectMeta
```

should work. however before we merge this, @bertinatto what was your concern about SC being deleted while PVC was being provisioned? Shouldn't the PVC be pending and start getting provisioned as soon as new SC comes back?

Reply:
I've used @gnufied's method to copy the old SC and examine the ObjectMeta. In addition, I have to check to see if the ReclaimPolicy is nil, as it's not defined in the `assets/*.yaml` file, and map this to the default value.

Reply:
> Shouldn't the PVC be pending and start getting provisioned as soon as new SC comes back?

I just tested this and yes, it behaves like you said. I think we're good re-creating the SC.